ehsm-service/server.js

```diff
@@ -0,0 +1,182 @@
+const express = require('express')
+const ffi = require('ffi-napi');
+const app = express();
+const logger = require('./logs/logger');
+const napiparams = require('./napiparams.js')
+
+app.use(express.json());
+
+const port = 3000;
+
+const result = (code, msg, data={}) => {
+  return {
+    code: code,
+    message: msg,
+    result: {
+      ...data
+    }
+  }
+}
+
+const apis= {
+  CreateKey: 'CreateKey',
+  Encrypt: 'Encrypt',
+  Decrypt: 'Decrypt',
+  GenerateDataKey: 'GenerateDataKey',
+  GenerateDataKeyWithoutPlaintext: 'GenerateDataKeyWithoutPlaintext',
+  ExportDataKey: 'ExportDataKey',
+  Sign: 'Sign',
+  Verify: 'Verify',
+}
+
+/**
+ * check sign
+ */
+const _checkSign = function (req,res,next){
+  const _logData = {
+    body: req.body,
+    query: req.query
+  }
+  logger.info(JSON.stringify(_logData));
+  const {appid,nonce,timestamp,sign, payload} = req.body;
+  if(!appid || !nonce|| !timestamp  || !sign || !payload) {
+    res.send(result(400,'Missing required parameters'));
+    return;
+  }
+  // todo sign 
+  next();
+}
+/**
+ * check payload
+ */
+ const _checkPayload = function (req,res,next){
+   const action = req.query.Action;
+  const { payload } = req.body;
+  const currentPayLoad = napiparams[action];
+  for (const key in currentPayLoad) {
+    if (payload[key] == undefined) {
+      res.send(result(400, 'The payload parameter is incomplete'));
+      return;
+    }
+    
+    // const p = currentPayLoad[key];
+    // for (const checkType in p) {
+    //   switch (checkType) {
+    //     case 'type':
+          
+    //       break;
+    //     case 'maxLength':
+        
+    //       break;
+    //     case 'length':
+        
+    //       break;
+    //     default:
+    //       break;
+    //   }
+    // }
+    // if()
+  }
+  next();
+}
+/**
+ * sign 
+ */
+
+app.use(_checkSign);
+/**
+ * payload 
+ */
+app.use(_checkPayload);
+/**
+ * parameter mechanism type
+ */
+const keyspecsType = {
+  "EHM_AES_GCM_128": 0,
```

Comment:
check the name define in ehsm_provider.h

Reply:
It will be fix.